cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -37,14 +37,14 @@
   private static final Gson GSON = new Gson();
 
   @Override
-  public void delete(Id.Namespace namespaceId) throws Exception {
-    URL url = resolve(String.format("unrecoverable/namespaces/%s", namespaceId.getId()));
+  public void delete(NamespaceId namespaceId) throws Exception {
```

Comment:
Let's go with Sree's view here.
But according to me we deprecated those few releases back giving sufficient time for people to switch to new APIs.